airflow/hooks/hive_hooks.py

```diff
@@ -122,11 +122,13 @@ def _prepare_cli_cmd(self):
                     template = utils.replace_hostname_pattern(
                         utils.get_components(template))
 
-                proxy_user = ""  # noqa
-                if conn.extra_dejson.get('proxy_user') == "login" and conn.login:
+                proxy_user = conn.extra_dejson.get('proxy_user', "")  # noqa
```

Comment:
actually.. sorry but what do you think of a _factory_ for the `proxy_user` like:
```
def _get_proxy_user(self):
    proxy_user_value = conn.extra_dejson.get('proxy_user', "")

    if proxy_user_value == "login" and conn.login:
        return "hive.server2.proxy.user={login}".format(login=conn.login)
    if proxy_user_value == "owner" and self.run_as:
        return "hive.server2.proxy.user={run_as}".format(run_as=self.run_as)
    if proxy_user_value != "":  # There is a custom proxy user
        return "hive.server2.proxy.user={proxy_user_value}".format(proxy_user_value=proxy_user_value)

     return proxy_user_value  # The default proxy user (undefined)
```
and then call `proxy_user = self._get_proxy_user()`

Reply:
@feluelle Sure. Done.